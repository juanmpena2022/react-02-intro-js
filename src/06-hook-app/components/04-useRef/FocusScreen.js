import { useRef } from "react";
import "../03-useFetch/effect.css";

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <>
      <h1>FocusScreen</h1>
      <hr />

      <input ref={inputRef} className="form-control" placeholder="name" />

      <button className="btn btn-primary mt-5" onClick={handleClick}>
        Focus
      </button>
    </>
  );
};
