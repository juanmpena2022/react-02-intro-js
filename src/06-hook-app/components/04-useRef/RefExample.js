import { useState } from "react";
import "../03-useFetch/effect.css";
import { MultipleCustomHooks } from "../03-useFetch/MultipleCustomHooks";

export const RefExample = () => {
  const [show, setShow] = useState(false);
  const handleShowHide = () => {
    setShow(!show);
  };

  return (
    <>
      <h1>RefExample</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button className="btn btn-primary" onClick={handleShowHide}>
        ShowHide
      </button>
    </>
  );
};
