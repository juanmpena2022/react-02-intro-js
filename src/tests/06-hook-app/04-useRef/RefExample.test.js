import { shallow } from "enzyme";
import { RefExample } from "../../../06-hook-app/components/04-useRef/RefExample";


describe("pruebas del componente <RefExample />", () => {
  test("snapshot", () => {
    const wrapper = shallow(<RefExample />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("debe mostrarse el componente MultipleCustomHooks luego del handle", () => {
    const wrapper = shallow(<RefExample />);
    wrapper.find('button').simulate('click');
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
