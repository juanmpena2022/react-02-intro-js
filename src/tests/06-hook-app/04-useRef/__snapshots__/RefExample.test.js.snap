// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente <RefExample /> snapshot 1`] = `
<Fragment>
  <h1>
    RefExample
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    ShowHide
  </button>
</Fragment>
`;
